jenkins/dependency-check.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+#
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Argument(s):
+#   ARTIFACT_FILE :  Artifact(groupId:artifactId:version:[[packaging]:classifier]) list file
+#
+# Used environment(s):
+#   SERVER_ID:      The repository id for this deployment.
+#   SERVER_URL:     The url where to deploy artifacts.
+#   M2_CACHE:       Maven local repo
+###
+
+set -ex
+
+ARTIFACT_FILE=${1:-"/tmp/artifacts-list"}
```

Comment:
artifact list file created by deploy.sh

Reply:
should 
```
com.nvidia:rapids-4-spark-parent_2.12:24.02.0-SNAPSHOT:pom
com.nvidia:rapids-4-spark-jdk-profiles_2.12:24.02.0-SNAPSHOT:pom
```
be in this list?


When we do the test with `mvn dependency:get`  for the documented artifacts such as `com.nvidia:rapids-4-spark_2.12:24.02.0-SNAPSHOT:jar:cuda11` and one of the pom-type artifacts  above is missing we should see a failure anyways, correct?

Reply:
@gerashegalov  Yes, you're right, jar's dependency contain the pom files! 

My original thought was to check anything we deployed, though pom check would be duplicated.


Let me remove pom from the check list.

Reply:
Updated, artifact list will be like

```
com.nvidia:rapids-4-spark_2.12:24.02.0-SNAPSHOT:jar
com.nvidia:rapids-4-spark_2.12:24.02.0-SNAPSHOT:jar:cuda11
com.nvidia:rapids-4-spark_2.12:24.02.0-SNAPSHOT:jar:cuda12
com.nvidia:rapids-4-spark_2.12:24.02.0-SNAPSHOT:jar:sources
com.nvidia:rapids-4-spark_2.12:24.02.0-SNAPSHOT:jar:javadoc
```

similar list for scala-2.13 jars, and for arm64 jars